### ISSUE TITLE
SC2128 is not triggered with read -ra stringarray <<<"${linevariable}"; Expanding an array without an index only gives the first element

### ISSUE THREAD

**Issue by evandrocoan** [koalaman/shellcheck#1636]
This code should trigger `SC2128: Expanding an array without an index only gives the first element`
```sh
#!/bin/bash
linevariable="var1     var2";
read -ra stringarray <<<"${linevariable}";
printf "first %s\\n" "${stringarray[0]}"
printf "second %s\\n" "${stringarray[1]}"

stringarray="${stringarray}"
printf "stringarray %s\\n" "${stringarray[0]}"
```

But only this triggers it:
```sh
#!/bin/bash
linevariable="var1     var2";
stringarray=(${linevariable});
printf "first %s\\n" "${stringarray[0]}"
printf "second %s\\n" "${stringarray[1]}"

stringarray="${stringarray}"
printf "stringarray %s\\n" "${stringarray[0]}"
```

Related to:
1. https://github.com/koalaman/shellcheck/issues/1630 SC2178 indefinitely triggers SC2128 when it should not, i.e., after overriding an array variable with a single string

**Comment by koalaman** [comment]
So it should! This failed because ShellCheck only looked for `-a` without accounting for things like `-ra`. This has now been fixed, thanks!
